HelloWorld/views.py

```diff
@@ -1,5 +1,67 @@
-from django.shortcuts import render
-from django.http import HttpResponse
-# Create your views here.
-def index(request):
-    return HttpResponse('Hello World!')
\ No newline at end of file
+from rest_framework import viewsets, filters
+from rest_framework.response import Response
+from rest_framework.decorators import action
+from django.utils import timezone
+from django.db import models
+from django_filters.rest_framework import DjangoFilterBackend
+from .models import Task, SubTask, Category
+from .serializers import TaskSerializer, SubTaskSerializer, CategorySerializer
+
+
+# ViewSet для задач
+class TaskViewSet(viewsets.ModelViewSet):
+    queryset = Task.objects.all()
+    serializer_class = TaskSerializer
+    filter_backends = [DjangoFilterBackend, filters.SearchFilter, filters.OrderingFilter]
+    filterset_fields = ['status', 'deadline']
+    search_fields = ['title', 'description']
+    ordering_fields = ['created_at']
+    ordering = ['created_at']
+
+    @action(detail=False, methods=['get'])
+    def statistics(self, request):
+        """ Статистика по задачам """
+        total_tasks = Task.objects.count()
+        status_counts = Task.objects.values('status').annotate(count=models.Count('status'))
+        overdue_tasks = Task.objects.filter(deadline__lt=timezone.now()).count()
+
+        return Response({
+            'total_tasks': total_tasks,
+            'status_counts': {item['status']: item['count'] for item in status_counts},
+            'overdue_tasks': overdue_tasks,
+        })
+
+    @action(detail=False, methods=['get'])
+    def tasks_by_day(self, request):
+        """ Получение задач по дню недели (пн=2, вт=3, ..., вс=1) """
+        day_of_week = request.query_params.get('day', None)
+        if day_of_week and day_of_week.isdigit():
+            queryset = Task.objects.filter(deadline__week_day=int(day_of_week))
+        else:
+            queryset = Task.objects.all()
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
+
+# ViewSet для подзадач
+class SubTaskViewSet(viewsets.ModelViewSet):
+    queryset = SubTask.objects.all()
+    serializer_class = SubTaskSerializer
+    filter_backends = [DjangoFilterBackend, filters.SearchFilter, filters.OrderingFilter]
+    filterset_fields = ['status', 'deadline']
+    search_fields = ['title', 'description']
+    ordering_fields = ['created_at']
+    ordering = ['created_at']
+
+
+# ViewSet для категорий
+class CategoryViewSet(viewsets.ModelViewSet):
+    queryset = Category.objects.all()
+    serializer_class = CategorySerializer
+
+    @action(detail=True, methods=['get'])
+    def count_tasks(self, request, pk=None):
+        category = self.get_object()
+        task_count = category.tasks.count()  # исправленный код
+        return Response({'category': category.name, 'task_count': task_count})
```

Comment:
Не уверен, но мне кажется, что так будет лучше
```suggestion
    @action(detail=False, methods=['get'])
    def count_tasks(self, request):
        stat = Category.objects.annotate(task_count=Count('task')).values('id', 'name', 'task_count')
        return Response(stat)
```